compiler/rustc_borrowck/src/type_check/mod.rs

```diff
@@ -357,12 +357,20 @@ impl<'a, 'b, 'tcx> Visitor<'tcx> for TypeVerifier<'a, 'b, 'tcx> {
                 .add_element(live_region_vid, location);
         });
 
+        // HACK(compiler-errors): Constants that are gathered into Body.required_consts
+        // have their locations erased...
+        let locations = if location != Location::START {
+            location.to_locations()
+        } else {
+            Locations::All(constant.span)
+        };
```

Comment:
It seems like you've identified the right course of action (modifying Body.required_consts instead of here).

*I*'d be ok with landing this as a temporary hack, but I'd rather getting sign off from somebody immediately related to the `const` generics effort. I'd punt to @oli-obk, but he'll be somewhat unavailable.

Reply:
I wonder if we should just unconditionally use the constant's span, it seems like that span will always be better than what we can guess from the location (or we'll just end up fetching the constant's span at the location anyway). At which point it begs the question of whether we should be passing a location to `visit_constant` at all. 

A quick skim of all `visit_constant` and `visit_const` suggest that we're only really interested in the span, not the actual mir location.

anyway. Leaving the hack comment and fixing this in a follow up PR seems the quickest way forward.

Reply:
Yeah, I looked into that refactoring, but we also use Location (and not Locations) as a key to some different HashMaps in different MIR stages? I'll see if I can give the refactoring another go, though.